py/server/deephaven/experimental/s3.py

```diff
@@ -118,12 +242,28 @@ def __init__(self,
                     (access_key_id is None and secret_access_key is not None)):
                 raise DHError("Either both access_key_id and secret_access_key must be provided or neither")
 
+            def throw_multiple_credentials_error(credentials1: str, credentials2: str):
+                raise DHError(f"Only one set of credentials can be set, but found {credentials1} and {credentials2}")
+
+            # Configure the credentials
             if access_key_id is not None:
                 if anonymous_access:
-                    raise DHError("Only one set of credentials may be used, requested both key and anonymous")
+                    throw_multiple_credentials_error("access_key_id", "anonymous_access")
+                if credentials is not None:
+                    throw_multiple_credentials_error("access_key_id", "credentials")
+                warn('access_key_id is deprecated, prefer setting credentials as '
```

Comment:
I would put the warning before the checks above just to make sure a user sees it.